exporters/metric/datadog/datadog.go

```diff
@@ -0,0 +1,178 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opentelemetry.io/otel/api/core"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+const (
+	// DefaultStatsAddrUDP specifies the default protocol (UDP) and address
+	// for the DogStatsD service.
+	DefaultStatsAddrUDP = "localhost:8125"
+)
+
+// NewExporter exports to a datadog client
+func NewExporter(opts Options) (*Exporter, error) {
+	if opts.StatsAddr == "" {
+		opts.StatsAddr = DefaultStatsAddrUDP
+	}
+	if opts.MetricNameFormatter == nil {
+		opts.MetricNameFormatter = defaultFormatter
+	}
+	client, err := statsd.New(opts.StatsAddr)
+	if err != nil {
+		return nil, err
+	}
+	return &Exporter{
+		client: client,
+		opts:   opts,
+	}, nil
+}
+
+// Options contains options for configuring the exporter.
+type Options struct {
+	// StatsAddr specifies the host[:port] address for DogStatsD. It defaults
+	// to localhost:8125.
+	StatsAddr string
+
+	// Tags specifies a set of global tags to attach to each metric.
+	Tags []string
+
+	// UseDistribution uses a DataDog Distribution type instead of Histogram
+	UseDistribution bool
+
+	// MetricNameFormatter lets you customize the metric name that gets sent to
+	// datadog before exporting
+	MetricNameFormatter func(namespace, name string) string
+}
+
+// Exporter forwards metrics to a DataDog agent
+type Exporter struct {
+	opts   Options
+	client *statsd.Client
+}
+
+const rate = 1
+
+func defaultFormatter(namespace, name string) string {
+	return name
+}
+
+func (e *Exporter) Export(ctx context.Context, cs export.CheckpointSet) error {
+	return cs.ForEach(func(r export.Record) error {
+		agg := r.Aggregator()
+		name := e.sanitizeMetricName(r.Descriptor().LibraryName(), r.Descriptor().Name())
+		itr := r.Labels().Iter()
+		tags := append([]string{}, e.opts.Tags...)
+		for itr.Next() {
```

Comment:
This has me wondering whether the `label.Set.Encoded()` method could be improved for exporters that don't use strings.  If you could cache a `[]string` in the label set, it would be a win here, but at present we only support caching string encodings.

Reply:
(No action required.)